containers/src/Kokkos_DynRankView.hpp

```diff
@@ -580,9 +580,38 @@ class DynRankView : private View<DataType*******, Properties...> {
              index_type i6 = 0) const {
     return view_type::operator()(i0, i1, i2, i3, i4, i5, i6);
   }
+
+// This is an accomodation for Phalanx, who use the operator[] to access all
+// memory
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
   KOKKOS_FUNCTION reference_type operator[](index_type i0) const {
+    if constexpr (std::is_same<typename drvtraits::value_type,
+                               typename drvtraits::scalar_array_type>::value) {
+      return view_type::data()[i0];
+    } else {
+      const size_t dim_scalar = view_type::impl_map().dimension_scalar();
+      const size_t bytes      = view_type::span() / dim_scalar;
+
+      using tmp_view_type = Kokkos::View<
+          DataType*, typename traits::array_layout,
+          typename traits::device_type,
+          Kokkos::MemoryTraits<traits::memory_traits::is_unmanaged |
+                               traits::memory_traits::is_random_access |
+                               traits::memory_traits::is_atomic>>;
+      tmp_view_type rankone_view(view_type::data(), bytes, dim_scalar);
+      return rankone_view(i0);
+    }
+  }
+#else
+  KOKKOS_FUNCTION reference_type operator[](index_type i0) const {
+#ifdef KOKKOS_ENABLE_DEBUG
+    if (view_type::rank() != 1u)
+      Kokkos::abort("DynRankView operator[] can only be used for rank-1");
+#endif
```

Comment:
This looks like a static property so we don't need a runtime check
```suggestion
    static_assert(view_type::rank() == 1u, "DynRankView operator[] can only be used for rank-1");
```
but it's also defined as
```
  using view_type = View<DataType*******, Properties...>;
```
so it wouldn't trigger anyway. What am I missing?

Reply:
yeah lol you are right it needs to call `rank()`

Reply:
```suggestion
#ifdef KOKKOS_ENABLE_DEBUG
    if (rank() != 1u)
      Kokkos::abort("DynRankView operator[] can only be used for rank-1");
#endif
```